src/DotLiquid/Util/NumericConverter.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace DotLiquid.Util
+{
+    /// <summary>
+    /// ExtensionMethods for converting to numeric values.
+    /// </summary>
+    public static class NumericConverter
+    {
+        private static bool IsReal(object o) => o is double || o is float || o is decimal;
+        private static bool IsInteger(object o) => o is int || o is uint || o is long || o is ulong || o is short || o is ushort || o is byte || o is sbyte;

```

Comment:
Shouldn't you have BigInteger here also?

Reply:
I don't think we support BigInteger anywhere.